compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -485,19 +485,21 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    TR_ASSERT(hasCompPtrs, "no compression sequence found under l2a node [%p]\n", node);
    */
 
-   // TODO (GuardedStorage)
-   TR::Register *source;
-   if (comp->useCompressedPointers() && (TR::Compiler->om.compressedReferenceShift() == 0 || firstChild->containsCompressionSequence()) && !node->isl2aForCompressedArrayletLeafLoad())
-      cg->setEvalCompressionSequence(true);
+   TR::Register* source = NULL;
 
-   if ((TR::Compiler->target.cpu.getS390SupportsGuardedStorageFacility()) && comp->useCompressedPointers() && (firstChild->containsCompressionSequence()) && !node->isl2aForCompressedArrayletLeafLoad())
+   // TODO (GuardedStorage): Why is isl2aForCompressedArrayletLeafLoad check necessary?
```

Comment:
This doesn't look like a straight-forward refactoring to me. It appears that the condition for compressedReferenceShift() == 0 has been eliminated, and consequently you've sunk the calls to setEvalCompressionSequence() so they no longer wrap around the code to evaluate the first child. Is there a reason the shift==0 condition is no longer required?

Side and probably irrelevant question because I have no idea what effect setEvalCompressionSequence() actually has. Is it impossible for there to be another l2a nested underneath the grandchild? On its face, it looks dangerous to me to simply setEvalCompressionSequence(false) on the way out if a nested l2a is possible, but there are lots of reasons why that could be perfectly correct.